apps/expo/src/app/filteredExercises/index.tsx

```diff
@@ -0,0 +1,190 @@
+
+export default function FilteredExercises(muscle = muscleSelectUser.NoSelection) {
+  const [addCount, setAddCount] = useState(0)
+  const { width: screenWidth, height: screenHeight } = Dimensions.get('window')
+  const { data, isFetched, isFetching } = api.exercise.getAllExercises.useQuery()
+  const [selectExercise, setSelect] = useState<{ [exerciseID: number]: boolean }>({})
+  const [] = useState()
+
+  // ------------------------------------------
+  // This is set to no selection for testing
+  // remove when everything done
+  muscle = muscleSelectUser.NoSelection
+  // ------------------------------------------
+
+  const selectToggle = (exerciseID: number) => {
+    setSelect((prevState) => ({
+      ...prevState,
+
+      [exerciseID]: !prevState[exerciseID],
+    }))
+
+    if (selectExercise[exerciseID]) {
+      // Post state true
+      if (addCount == 0) {
+        setAddCount(1)
+      } else {
+        setAddCount(addCount + 1)
+      }
+    } else {
+      // Post state false
+      if (addCount == 1) {
+        setAddCount(0)
+      } else {
+        // Negative value should not appear
+        setAddCount(addCount - 1)
+      }
+    }
+  }
+
+  return (
+    <SafeAreaView style={{ backgroundColor: '#1C1B1B', flex: 1 }}>
+      {/*<NavBar*/}
+      {/*  center={'Select Exercises'}*/}
+      {/*  right={*/}
+      {/*    <Pressable onPress={() => router.back()}>*/}
+      {/*      {addCount == 0 ? (*/}
+      {/*        <Text style={{ color: '#CACACA', fontSize: 20 }}>Add</Text>*/}
+      {/*      ) : (*/}
+      {/*        <Text style={{ color: '#CACACA', fontSize: 20 }}>Add{addCount}</Text>*/}
+      {/*      )}*/}
+      {/*    </Pressable>*/}
+      {/*  }*/}
+      {/*/>*/}
+
+      <View className='flex flex-row justify-between px-5'>
+        <Ionicons onPress={() => router.back()} name='chevron-back' size={20} color='#CACACA' />
+        <Text style={{ color: '#CACACA', fontSize: 20 }}>Select exercise</Text>
+        <Pressable onPress={() => router.back()}>
+          {addCount == 0 ? (
+            <Text style={{ color: '#CACACA', fontSize: 20 }}>Add</Text>
+          ) : (
+            <Text style={{ color: '#CACACA', fontSize: 20 }}>Add{addCount}</Text>
+          )}
+        </Pressable>
+      </View>
+
+      {/*<View className='flex flex-row items-center'>*/}
+      {/*  <Image*/}
+      {/*    source={require('~assets/award/trophy.png')}*/}
+      {/*    style={{*/}
+      {/*      flex: 1,*/}
+      {/*      width: screenWidth * 0.9,*/}
+      {/*      height: screenWidth * 0.9,*/}
+      {/*    }}*/}
+      {/*    resizeMode='contain'*/}
+      {/*  />*/}
+      {/*</View>*/}
+
+      <View className='pb-px-20 flex flex-row' style={{ height: 0.7 * screenHeight - 50 }}>
+        <ScrollView>
+          {data?.map((exercise: any) => (
+            <View key={exercise.id} className='flex flex-row rounded-lg p-2'>
+              {exercise?.bodyPart == 'ARMS' && muscle == muscleSelectUser.Arm && (
```

Comment:
From here to line 161 could be refactored into a loop, but I understand if we don't have time for that now